src/main/java/org/fenixedu/academic/domain/accessControl/AlumniGroup.java

```diff
@@ -96,20 +122,46 @@ public boolean isMember(User user, DateTime when) {
         if (user == null || user.getPerson().getStudent() == null || user.getPerson().getStudent().getAlumni() == null) {
             return false;
         }
-        if (degree != null) {
-            for (Registration registration : user.getPerson().getStudent().getRegistrationsFor(degree)) {
-                if (registration.isRegistrationConclusionProcessed()) {
-                    return true;
-                }
+        if (registered != null) {
+            if (registered) {
+                return isRegistered(user, when);
+            } else if (!registered) {
+                return !isRegistered(user, when);
             }
-            return false;
+        }
+        return isAlumni(user.getPerson().getStudent());
+    }
+
+    private boolean isRegistered(User user, DateTime when) {
```

Comment:
According to the issue the expected semantic of registered is:
Alumnus that fulfilled the alumni registration process, not students that have concluded a given degree